WORKSPACE

```diff
@@ -0,0 +1,337 @@
+        "https://repo1.maven.org/maven2/org/ow2/asm/asm-analysis/7.0/asm-analysis-7.0.jar",
+    ],
+    exports = [
+        "@org_ow2_asm",
+        "@org_ow2_asm_tree",
+    ],
+)
+
+# BSD 3-clause
+java_import_external(
+    name = "org_ow2_asm_commons",
+    jar_sha256 = "fed348ef05958e3e846a3ac074a12af5f7936ef3d21ce44a62c4fa08a771927d",
+    jar_urls = [
+        "https://mirror.bazel.build/repo1.maven.org/maven2/org/ow2/asm/asm-commons/7.0/asm-commons-7.0.jar",
+        "https://repo1.maven.org/maven2/org/ow2/asm/asm-commons/7.0/asm-commons-7.0.jar",
+    ],
+    exports = ["@org_ow2_asm_tree"],
+)
+
+# BSD 3-clause
+java_import_external(
+    name = "org_ow2_asm_tree",
+    jar_sha256 = "cfd7a0874f9de36a999c127feeadfbfe6e04d4a71ee954d7af3d853f0be48a6c",
+    jar_urls = [
+        "https://mirror.bazel.build/repo1.maven.org/maven2/org/ow2/asm/asm-tree/7.0/asm-tree-7.0.jar",
+        "https://repo1.maven.org/maven2/org/ow2/asm/asm-tree/7.0/asm-tree-7.0.jar",
+    ],
+    exports = ["@org_ow2_asm"],
+)
+
+# BSD 3-clause
+java_import_external(
+    name = "org_ow2_asm_util",
+    jar_sha256 = "75fbbca440ef463f41c2b0ab1a80abe67e910ac486da60a7863cbcb5bae7e145",
+    jar_urls = [
+        "https://mirror.bazel.build/repo1.maven.org/maven2/org/ow2/asm/asm-util/7.0/asm-util-7.0.jar",
+        "https://repo1.maven.org/maven2/org/ow2/asm/asm-util/7.0/asm-util-7.0.jar",
+    ],
+    exports = [
+        "@org_ow2_asm_analysis",
+        "@org_ow2_asm_tree",
+    ],
+)
+
+# MIT-style license
+java_import_external(
+    name = "org_json",
+    jar_sha256 = "0aaf0e7e286ece88fb60b9ba14dd45c05a48e55618876efb7d1b6f19c25e7a29",
+    jar_urls = [
+        "https://mirror.bazel.build/repo1.maven.org/maven2/org/json/json/20160212/json-20160212.jar",
+        "https://repo1.maven.org/maven2/org/json/json/20160212/json-20160212.jar",
+        "http://maven.ibiblio.org/maven2/org/json/json/20160212/json-20160212.jar",
+    ],
+)
+
+# Apache 2.0
+http_archive(
+    name = "rules_java",
+    sha256 = "52423cb07384572ab60ef1132b0c7ded3a25c421036176c0273873ec82f5d2b2",
+    urls = [
+        "https://github.com/bazelbuild/rules_java/releases/download/0.1.0/rules_java-0.1.0.tar.gz",
+    ],
+)
+load("@rules_java//java:repositories.bzl", "rules_java_dependencies", "rules_java_toolchains")
+rules_java_dependencies()
+rules_java_toolchains()
+
+# Apache 2.0
+http_archive(
+    name = "rules_proto",
+    sha256 = "57001a3b33ec690a175cdf0698243431ef27233017b9bed23f96d44b9c98242f",
+    strip_prefix = "rules_proto-9cd4f8f1ede19d81c6d48910429fe96776e567b1",
+    urls = [
+        "https://github.com/bazelbuild/rules_proto/archive/9cd4f8f1ede19d81c6d48910429fe96776e567b1.tar.gz",
+    ],
+)
+load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")
+rules_proto_dependencies()
+rules_proto_toolchains()
+
+# Apache 2.0
+http_archive(
+    name = "xyz_yannic_rules_javacc",
```

Comment:
what does this do?

Reply:
It's used in `soyparse` to generate Java files from `SoyFileParser.jj` with [javacc](https://javacc.org) and compile them together with the other Java sources in that package.